ipmininet/node_description.py

```diff
@@ -48,9 +48,9 @@ def addDaemon(self, daemon: Union[Daemon, Type[Daemon]],
 
 
 class OpenrRouterDescription(RouterDescription):
-    def addDaemon(self, daemon: Union[OpenrDaemon, Type[OpenrDaemon]],
-                  default_cfg_class: Type[OpenrConfig] = OpenrConfig,
-                  **kwargs):
+    def addOpenrDaemon(self, daemon: Union[OpenrDaemon, Type[OpenrDaemon]],
```

Comment:
I don't understand
Why not leaving the ``addDaemon`` and downgrading ``OpenrDaemon`` to ``Daemon`` ?

Reply:
> I don't understand
> Why not leaving the `addDaemon` and downgrading `OpenrDaemon` to `Daemon` ?

This is already done by inheritance, right ?

Reply:
`addOpenrDaemon` is just a convenience method that helps to add `OpenrDaemon`s to `OpenrRouter`s.

Reply:
Maybe I should replace ~`super().addDaemon(...)`~ with `self.addDaemon(...)`

Reply:
>  >  I don't understand
    Why not leaving the addDaemon and downgrading OpenrDaemon to Daemon ?

> This is already done by inheritance, right ?

If your point is to use OpenrConfig by default, you need to redefine addDaemon, otherwise inheritance won't do it for you.

Moreover, a configuration object is by node and not by daemon, so your OpenrConfig should be compatible with the other daemons as well. And if this is the case, there is no reason for not using it to addDaemon.

Reply:
> > > I don't understand
> > > Why not leaving the addDaemon and downgrading OpenrDaemon to Daemon ?
> 
> > This is already done by inheritance, right ?
> 
> If your point is to use OpenrConfig by default, you need to redefine addDaemon, otherwise inheritance won't do it for you.

I got your point, not to overwrite `addDaemon` this is why I do not want to touch the inherited implementation. I just renamed the method to `addOpenrDaemon` to provide a method that makes it easy to add `Openr` Daemons to an `OpenrRouter`.

> Moreover, a configuration object is by node and not by daemon, so your OpenrConfig should be compatible with the other daemons as well. And if this is the case, there is no reason for not using it for addDaemon.

`OpenrConfig` is a bad name then, since it implies relation to the Daemon class. `OpenrRouterConfig` should be more appropriate though.